src/hotspot/share/runtime/threadHeapSampler.cpp

```diff
@@ -356,7 +356,7 @@ double ThreadHeapSampler::fast_log2(const double& d) {
   assert(sizeof(d) == sizeof(x),
          "double and uint64_t do not have the same size");
   x = *reinterpret_cast<const uint64_t*>(&d);
-  const uint32_t x_high = x >> 32;
+  const uint32_t x_high = (uint32_t)(x >> 32);
```

Comment:
Can't we do a checked_cast here? The cast should be reversible with that shift value.

Reply:
I think so.  I was afraid of sign extension so I just wanted to chop it off, also didn't seem necessary to check.